app/Http/Controllers/ProductController.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Services\CSVService;
+use App\Services\ProductService;
+
+class ProductController extends Controller
+{
+    private $csvService, $productService;
+
+    public function __construct(CSVService $csvService, ProductService $productService)
+    {
+        $this->csvService = $csvService;
+        $this->productService = $productService;
+    }
```

Comment:
https://php.watch/versions/8.0/constructor-property-promotion
Could be refactored as: 

```suggestion
public function __construct(private CSVService $csvService, private ProductService $productService) {}
```